launch_ros/launch_ros/actions/node.py

```diff
@@ -419,6 +419,30 @@ def _perform_substitutions(self, context: LaunchContext) -> None:
         self.__final_node_name = prefix_namespace(
             self.__expanded_node_namespace, self.__expanded_node_name)
 
+        # Check to see if a global parameter file was provided
+        # First extract the raw name of the node
+        node_name_as_text = None
+        if type(self.__node_name) == list:
+            node_name_as_text = list(self.__node_name[0].__dict__.values())[0]
+        elif type(self.__node_name) == str:
+            node_name_as_text = self.__node_name
```

Comment:
this is not the correct way of using substitutions, I think `self.__final_node_name` in L419 is already what you want

Reply:
Also, support for namespaces to be added